### PR TITLE
Ensure the R help html file to display in the Maximise window

### PR DESCRIPTION
**Pull request by N-thony**
@lloyddewit I have made the change as requested. Could you have a look?
@rdstern I can help you to test this. There is a small thing to do before as I discussed with @lloyddewit 
@Patowhiz you can also have a look.

### CODE REVIEW

File: instat/dlgFromLibrary.vb [hunk 1]
```diff
@@ -229,13 +229,34 @@ Public Class dlgFromLibrary
...
-        clsHelp.SetRCommand("help")
-        clsHelp.AddParameter("topic", Chr(34) & lstCollection.SelectedItems(0).Text & Chr(34))
-        clsHelp.AddParameter("package", Chr(34) & ucrInputPackages.cboInput.SelectedItem & Chr(34))
-        clsHelp.AddParameter("help_type", Chr(34) & "html" & Chr(34))
-        frmMain.clsRLink.RunScript(clsHelp.ToScript, strComment:="Opening help page for" & " " & lstCollection.SelectedItems(0).Text & " " & "dataset. Generated from dialog Open Dataset from Library", iCallType:=2, bSeparateThread:=False, bUpdateGrids:=False)
+        Dim strTopic As String = lstCollection.SelectedItems(0).Text
+        Dim strPackage As String = ucrInputPackages.cboInput.SelectedItem
+
+        Dim strTempFile As String = String.Concat(strTopic, ".html")
+        Dim strTempPath As String = "C:\Program Files\R-Instat\0.7.8\static"
```
Comment (lloyddewit) [inline instat/dlgFromLibrary.vb:236]: This variable is only used in the commented testing line below. Remove?
```suggestion
```

File: instat/dlgFromLibrary.vb [hunk 2]
```diff
@@ -229,13 +229,34 @@ Public Class dlgFromLibrary
...
+        Dim strHelpPath As String = String.Concat(frmMain.strStaticPath, strRLibrary, strTempFile)
+
+        If strPackage <> "datasets" Then
+            strTempFile = "00Index.html"
+            strRLibrary = String.Concat("\R\library\", strPackage, "\html\")
+            strHelpPath = String.Concat(frmMain.strStaticPath, strRLibrary, strTempFile)
+        End If
+
+        'use this line to test in the development mode
+        'strHelpPath = String.Concat(strTempPath, strRLibrary, strTempFile)
```
Comment (lloyddewit) [inline instat/dlgFromLibrary.vb:247]:
```suggestion
        'if you wish to test this in the development environment, then uncomment line below
        '(you may need to change the path or version number)
        'strHelpPath = String.Concat("C:\Program Files\R-Instat\0.7.8\static", strRLibrary, strTempFile)
```

File: instat/dlgModelling.vb [hunk 3]
```diff
@@ -582,15 +582,25 @@ Public Class dlgModelling
...
-        Dim clsHelp As New RFunction
-        Dim strPackageName As String
-
-        strPackageName = ucrInputComboRPackage.GetText
-        clsHelp.SetPackageName("utils")
-        clsHelp.SetRCommand("help")
-        clsHelp.AddParameter("package", Chr(34) & strPackageName & Chr(34))
-        clsHelp.AddParameter("help_type", Chr(34) & "html" & Chr(34))
-        frmMain.clsRLink.RunScript(clsHelp.ToScript, strComment:="Opening help page for" & " " & strPackageName & " " & "Package. Generated from dialog Modelling", iCallType:=2, bSeparateThread:=False, bUpdateGrids:=False)
+        Dim strPackageName As String = ucrInputComboRPackage.GetText
```
Comment (lloyddewit) [inline instat/dlgModelling.vb:585]: this code seems very similar to the code starting on line 232.
Make a new sub and call this sub from 3 places?

File: instat/ucrCalculator.vb [hunk 4]
```diff
@@ -1497,16 +1497,28 @@ Public Class ucrCalculator
 
 
     Private Sub OpenHelpPage()
-        Dim clsHelp As New RFunction
+        Dim strTempFile As String = "\html\00Index.html"
```
Comment (lloyddewit) [inline instat/ucrCalculator.vb:1500]: this code seems very similar to the code starting on line 232.
Make a new sub and call this sub from 3 places?

**Review by lloyddewit — CHANGES_REQUESTED** [review]

**Review by rdstern — COMMENTED** [review]
@N-thony I tried the R Help from the File > Open from library and it still displays in a browser.  And the calculator, and the package help.  I didn't see anything in the maximized window.  Am I doing something wrong?

**Comment by N-thony** [comment]
> @N-thony I tried the R Help from the File > Open from library and it still displays in a browser. And the calculator, and the package help. I didn't see anything in the maximized window. Am I doing something wrong?

@rdstern yes, I can help on how to test it.

**Review by lloyddewit — DISMISSED** [review]
@N-thony Thank you for the changes and discussions, happy to approve

**Comment by lloyddewit** [comment]
@rdstern if you can test/approve, then we can merge.
Note: to test this PR in your development environment, you'll need to uncomment 3 lines (1 line in dlgModeling, ucrCalculator and dlgFromLibrary respectively). If you look at the code changes in this PR, then it should be obvious which lines need uncommenting.
thanks

File: instat/ucrCalculator.vb [hunk 5]
```diff
@@ -1497,16 +1499,41 @@ Public Class ucrCalculator
 
 
     Private Sub OpenHelpPage()
-        Dim clsHelp As New RFunction
+        Dim strLocalHost As String = "127.0.0.1"
+        Dim clsStartServer As New RFunction
+        Dim clsGetPort As New RFunction
+        Dim clsList As New RFunction
```
Comment (lloyddewit) [inline instat/ucrCalculator.vb:1505]: use normal naming?
```suggestion
        Dim clsStartServerFunction As New RFunction
        Dim clsGetPortfunction As New RFunction
        Dim clsListFunction As New RFunction
```

**Review by lloyddewit — CHANGES_REQUESTED** [review]
Just one suggestion. When this is resolved (and the testing lines are recommented), then I'm happy to approve.

**Comment by rdstern** [comment]
@N-thony after our discussion I now understand that it can't work visibly in the development version, or even in the merged version.  So, as soon as @lloyddewit is ok, then I am happy for it to be merged.  I note the places we currently have the calls to the R-help:

a) The File > Open from Library
b) Each halfway dialogue - currently 4, namely Prepare > Calculator, Model > Fit > Fit Model Keyboard, and Hypothesis Testing keyboard, and Use Model > Use Model Keyboard.
I expect more in the future.
c) Help > Packages Documentation.  (When frm_main is next edited, then this could change to R Packages Documentation.

There are also many calls to the packages from the help.  I am thinking this may remain as an online call as I call the CRAN page where possible.

**Comment by N-thony** [comment]
> @N-thony after our discussion I now understand that it can't work visibly in the development version, or even in the merged version. So, as soon as @lloyddewit is ok, then I am happy for it to be merged. I note the places we currently have the calls to the R-help:
> 
> a) The File > Open from Library b) Each halfway dialogue - currently 4, namely Prepare > Calculator, Model > Fit > Fit Model Keyboard, and Hypothesis Testing keyboard, and Use Model > Use Model Keyboard. I expect more in the future. c) Help > Packages Documentation. (When frm_main is next edited, then this could change to R Packages Documentation.
> 
> There are also many calls to the packages from the help. I am thinking this may remain as an online call as I call the CRAN page where possible.

@rdstern this first solution is not really efficient, I found a solution that will works well anyways but the code runs well in the script. I'm investigating on it and will also check with @lloyddewit

**Pull request closed without merging**